src/solvers/dgsem_t8code/containers_3d.jl

```diff
@@ -0,0 +1,232 @@
+            element = t8_forest_get_element_in_tree(mesh.forest, itree, ielement)
+            element_level = t8_element_level(eclass_scheme, element)
+
+            element_length = t8_hex_len(element_level) / t8_hex_root_len
+
+            element_coords = Vector{Cdouble}(undef, 3)
+            t8_element_vertex_reference_coords(eclass_scheme, element, 0,
+                                               pointer(element_coords))
+
+            nodes_out_x = 2 *
+                          (element_length * 1 / 2 * (nodes .+ 1) .+ element_coords[1]) .-
+                          1
+            nodes_out_y = 2 *
+                          (element_length * 1 / 2 * (nodes .+ 1) .+ element_coords[2]) .-
+                          1
+            nodes_out_z = 2 *
+                          (element_length * 1 / 2 * (nodes .+ 1) .+ element_coords[3]) .-
+                          1
+
+            polynomial_interpolation_matrix!(matrix1, mesh.nodes, nodes_out_x,
+                                             baryweights_in)
+            polynomial_interpolation_matrix!(matrix2, mesh.nodes, nodes_out_y,
+                                             baryweights_in)
+            polynomial_interpolation_matrix!(matrix3, mesh.nodes, nodes_out_z,
+                                             baryweights_in)
+
+            multiply_dimensionwise!(view(node_coordinates, :, :, :, :,
+                                         current_index += 1),
+                                    matrix1, matrix2, matrix3,
+                                    view(mesh.tree_node_coordinates, :, :, :, :,
+                                         itree + 1),
+                                    tmp1)
+        end
+    end
+
+    return node_coordinates
+end
+
+# This routine was copied and adapted from `src/dgsem_p4est/containers_3d.jl`: `orientation_to_indices_p4est`.
+function trixi_t8_init_mortar_neighbor_ids!(mortars::P4estMortarContainer{3}, my_face,
+                                            other_face, orientation, neighbor_ielements,
+                                            mortar_id)
+    # my_face and other_face are the face directions (zero-based)
+    # of "my side" and "other side" respectively.
+    # Face corner 0 of the face with the lower face direction connects to a corner of the other face.
+    # The number of this corner is the orientation code in `p4est`.
+    lower = my_face <= other_face
+
+    # x_pos, y_neg, and z_pos are the directions in which the face has right-handed coordinates
+    # when looked at from the outside.
+    my_right_handed = my_face in (1, 2, 5)
+    other_right_handed = other_face in (1, 2, 5)
+
+    # If both or none are right-handed when looked at from the outside, they will have different
+    # orientations when looked at from the same side of the interface.
+    flipped = my_right_handed == other_right_handed
+
+    # In the following illustrations, the face corner numbering of `p4est` is shown.
```

Comment:
Is this corner numbering the same in `t8code`?

Reply:
Yes. T8code uses the same corner numbering convention.